crates/transaction/reveal/src/visitor.rs

```diff
@@ -0,0 +1,66 @@
+use strata_primitives::hash;
+use strata_state::{batch::SignedBatchCheckpoint, da_blob::PayloadCommitment};
+
+pub trait EnvelopeVisitor<'t> {
+    type Type;
+    fn visit_envelope_payload_chunks(chunks: impl Iterator<Item = &'t [u8]>) -> Self::Type;
+}
+// CheckpointVisitor
+pub trait CheckpointVisitor<'t>: EnvelopeVisitor<'t> {
+    fn visit_checkpoint(&self) -> Option<SignedBatchCheckpoint>;
+}
+
+#[derive(Debug, Clone)]
+pub struct CommitmentVisitor {
+    commitment: PayloadCommitment,
+}
+
+impl CommitmentVisitor {
+    pub fn commitment(&self) -> PayloadCommitment {
+        self.commitment
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ExtractorVisitor<'a> {
+    chunks: Vec<&'a [u8]>,
```

Comment:
Would be a `Vec<Vec<u8>>` or similar, as we discussed in DMs.  We coalese the chunks into a single entry since they're semantically contiguous.